### PR TITLE
chore(typescript): Replace React.FC and React.PropsWithChildren

### PR DESCRIPTION
**Pull request by jonasbuechel**
# Get types ready for React 18
@types/react at version 18 does not have implicit children property anymore so we decided to replace React.FC and Reac.PropsWithChildren with simple type definitions.

## Replacing React.FC
**Before:** 
`const MyComponent: FC<MyComponentProps> = ({ someProperty }) => ()`

**After:** (children only in use where needed)
`const MyComponent = ({ someProperty, children }: MyComponentProps): ReactElement => ()`

## Replacing PropsWithChildren
**Before:**
`export type MyComponentProps = PropsWithChildren<{
    emphasis?: Emphasis;
    size?: Size;
}>;`

**After:**
```
export type MyComponentProps = {
    emphasis?: Emphasis;
    size?: Size;
    children?: ReactNode;
};

```

### CODE REVIEW

File: src/components/Breadcrumbs/Breadcrumbs.cy.tsx [hunk 1]
```diff
@@ -20,16 +20,19 @@ const BREADCRUMB_ITEMS_MIXED_ELEMENTS = [
     { label: 'item with link', link: '/some-third-link' },
 ];
 
-const ChangingBreadcrumbs: FC<BreadcrumbsProps> = () => {
-    const [items, setItems] = useState(BREADCRUMB_ITEMS);
+const ChangingBreadcrumbs = ({ items }: BreadcrumbsProps): ReactElement => {
```
Comment (SamuelAlev) [inline src/components/Breadcrumbs/Breadcrumbs.cy.tsx:23]: Why the change?

**Review by SamuelAlev — APPROVED** [review]
Small details comments, feel free to merge

File: src/components/Accordion/types.ts [hunk 2]
```diff
@@ -17,14 +17,14 @@ export type AriaAccordionItemProps = {
...
-    headerComponent?: (props: AccordionHeaderProps) => ReactElement;
+    headerComponent?: (props: AccordionHeaderProps) => ReactElement | null;
 };
 
 export type AccordionItemProps = PropsWithChildren<{
     header: Omit<AccordionHeaderProps, 'isOpen'> & { active?: boolean; onClick?: () => void };
     padding?: boolean;
     divider?: boolean;
-    headerComponent?: (props: AccordionHeaderProps) => ReactElement;
+    headerComponent?: (props: AccordionHeaderProps) => ReactElement | null;
```
Comment (SamuelAlev) [inline src/components/Accordion/types.ts:27]: Not needed anymore

**Review by SamuelAlev — CHANGES_REQUESTED** [review]

File: src/components/Accordion/types.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
 import { TreeState } from '@react-stately/tree';
-import { FC, PropsWithChildren, ReactNode } from 'react';
+import { PropsWithChildren, ReactElement, ReactNode } from 'react';
```
[the remark this answers is not in the document]
Comment (jonasbuechel) [reply src/components/Accordion/types.ts]: removed then all with https://github.com/Frontify/fondue/commit/55f5fcd1e2ede8d450979426b93b15aa789f4b45

File: src/components/Breadcrumbs/Breadcrumbs.cy.tsx (hunk 1 above)
Comment (jonasbuechel) [reply src/components/Breadcrumbs/Breadcrumbs.cy.tsx:23]: Needed to correctly type the function and now needed to replace the already used prop `items`, which was in use but also passed BREADCRUM_ITEMS 😸

File: src/components/Accordion/types.ts (hunk 2 above)
Comment (SamuelAlev) [reply src/components/Accordion/types.ts:27]: You can update the Clarify code to comply, Fondue shouldn't adapt to Clarify (this is important)

Comment (fabriciomendonca) [reply src/components/Accordion/types.ts:27]: This is the same discussion we are having here: https://github.com/Frontify/fondue/pull/1340
I'm using `JSX.Element` which is the default inferred type by Typescript when you create a component without `React.FC`

Comment (fabriciomendonca) [reply src/components/Accordion/types.ts:27]: Changes on Clarify in this PR https://github.com/Frontify/clarify/pull/14480

Comment (jonasbuechel) [reply src/components/Accordion/types.ts:27]: Fair point, requires those changes to be released first
https://github.com/Frontify/clarify/pull/14562

Comment (jonasbuechel) [reply src/components/Accordion/types.ts]: removed them 👍

**Review by imoutaharik — COMMENTED** [review]
Thanks for taking care! 🥇 
I did a smoke test and didn't find anything breaking in Clarify.

Just left some small comments.

File: scripts/templates/component.ts [hunk 4]
```diff
@@ -6,9 +6,9 @@ import { toKebabCase } from '../transforms';
 export const setup = (componentName: string): ComponentFileBuilderResponse => ({
     content: `/* (c) Copyright Frontify Ltd., all rights reserved. */
 
-import React, { PropsWithChildren } from "react";
+import React, { ReactNode } from "react";
 
-export type ${componentName}Props = PropsWithChildren<{ foo: string }>;
+export type ${componentName}Props = { foo: string; children?: ReactNode; };
```
[the remark this answers is not in the document]
Comment (jonasbuechel) [reply scripts/templates/component.ts]: done 👍

File: src/components/MultiInput/MultiInput.tsx [hunk 5]
```diff
@@ -10,9 +10,10 @@ export enum MultiInputLayout {
 export type MultiInputProps = {
     layout: MultiInputLayout;
     spanLastItem?: boolean;
+    children: ReactNode;
```
[the remark this answers is not in the document]
Comment (jonasbuechel) [reply src/components/MultiInput/MultiInput.tsx:13]: Yes, the component won't have any output without children, right?

[the remark this answers is not in the document]
Comment (imoutaharik) [reply src/components/MultiInput/MultiInput.tsx:13]: Yes, that's right. This component needs children for the logic inside.

**Review by imoutaharik — APPROVED** [review]
